lib/workers/bin/run_single_worker.rb

```diff
@@ -106,12 +106,18 @@ def all_role_names
     runner_options[:ems_id]     = options[:ems_id].length == 1 ? options[:ems_id].first : options[:ems_id].collect { |id| id }
   end
 
+  update_options = create_options.dup
+  # If a guid is provided, raise if it's not found, update otherwise
+  # Because podified needs to create on the first run_single_worker and update after:
+  #  If system_uid is provided, update if found, create if not found.
+  #  TODO:  This is really inconsistent and confusing.  Why can't GUID follow the same rules?
```

Comment:
:+1: for GUID following the same rules.  It isn't being created currently because we expect MiqServer to create it prior to starting the worker on process and systemd runtimes but it wouldn't take much change to go through `create_worker_record` here in all cases (we'll have to pass in EMS_ID in the systemd conf file as this is currently looked-up by in the worker record and not passed in)

Reply:
Yeah, I put this here because I didn't think it made sense to change the GUID behavior but I wanted it to be clear that GUID and sytem_uid are now even more different than before as you can't really tell why from the code.